core/storage/learner_group/gae_models.py

```diff
@@ -276,16 +278,67 @@ def apply_deletion_policy(cls, user_id: str) -> None:
                 learner_group_model.put()
 
             # If the user is the facilitator of the group and there are
-            # more then one facilitators_user_ids, delete the user from the
-            # facilitators_user_ids list.
-            elif user_id in learner_group_model.facilitators_user_ids and (
-                    len(learner_group_model.facilitators_user_ids) > 1):
-                learner_group_model.facilitators_user_ids.remove(user_id)
+            # more then one facilitator_user_ids, delete the user from the
+            # facilitator_user_ids list.
+            elif user_id in learner_group_model.facilitator_user_ids and (
+                    len(learner_group_model.facilitator_user_ids) > 1):
+                learner_group_model.facilitator_user_ids.remove(user_id)
                 learner_group_model.update_timestamps()
                 learner_group_model.put()
 
             # If the user is the facilitator of the group and there is
             # only one facilitator_user_id, delete the group.
-            elif user_id in learner_group_model.facilitators_user_ids and (
-                    len(learner_group_model.facilitators_user_ids) == 1):
+            elif user_id in learner_group_model.facilitator_user_ids and (
+                    len(learner_group_model.facilitator_user_ids) == 1):
+                learner_group_model.delete_learner_group_references(
+                    learner_group_model.id)
                 learner_group_model.delete()
+
+    @classmethod
```

Comment:
The function looks good, though shouldn’t it be a LearnerGroupUser model function instead of for LearnerGroupModel?

Reply:
Done!